### PR TITLE
The purge & restructure PART I

### PR DESCRIPTION
**Pull request by ouchadam**
First step to modularising notils

- Converts notils to a android library 
- Creates two main packages, `android` & `java` (these will become our basic modules)
- Drops the `Cursor`, `Animation`, `Adapter`, `Fragment` & `View Server` packages

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,28 +1,29 @@
...
-        classpath 'com.novoda:bintray-release:0.2.4'
+        classpath 'com.android.tools.build:gradle:1.5.0'
+        classpath 'com.novoda:bintray-release:0.2.7'
     }
 }
 
-apply plugin: 'propdeps'
-apply plugin: 'propdeps-maven'
-apply plugin: 'propdeps-idea'
-apply plugin: 'propdeps-eclipse'
```
Comment (ataulm) [inline build.gradle]: :+1: lol?

File: build.gradle [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
+    jcenter()
 }
 
-dependencies {
-    optional 'com.google.android:android:4.0.1.2'
-    optional 'com.google.android:support-v4:r7'
+android {
+    defaultConfig {
+        minSdkVersion 15
+        targetSdkVersion 22
```
Comment (ataulm) [inline build.gradle]: target 23? I'm not certain this affects apps.

File: build.gradle (hunk 1 above)
Comment (ouchadam) [inline build.gradle]: they were all added to be able to use `optional` dependencies

File: src/main/java/com/novoda/notils/android/caster/Fragments.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    private Fragments() {
+    }
+
+    @SuppressWarnings("unchecked")
+    public static <T> T findFragmentById(android.app.FragmentManager fragmentManager, int fragmentId) {
+        return (T) fragmentManager.findFragmentById(fragmentId);
+    }
+
+    @SuppressWarnings("unchecked")
+    public static <T> T findFragmentByTag(android.app.FragmentManager fragmentManager, String fragmentTag) {
```
Comment (ataulm) [inline src/main/java/com/novoda/notils/android/caster/Fragments.java]: Didn't this used to have other versions for the support fragments?

File: src/main/java/com/novoda/notils/android/logger/Log.java [hunk 4]
```diff
@@ -201,7 +201,7 @@ private static String formatString(Object... msg) {
      */
     public static void setShowLogs(boolean showLogs) {
         Log.INITIALISED = true;
-        Log.SHOW_LOGS = showLogs;
+        SHOW_LOGS = showLogs;
```
Comment (ataulm) [inline src/main/java/com/novoda/notils/android/logger/Log.java]: pfft dropped the prefix for this but not the one above?

Comment (ataulm) [inline src/main/java/com/novoda/notils/android/logger/Log.java]: this shouldn't be uppercase - it's not a constant

File: src/main/java/com/novoda/notils/caster/Fragments.java [hunk 5]
```diff
@@ -1,31 +0,0 @@
...
-    /**
-     * Simpler version of {@link android.support.v4.app.FragmentManager#findFragmentById(int)}} which infers the target type.
-     */
-    @SuppressWarnings("unchecked")
-    public static <T> T findFragmentById(android.support.v4.app.FragmentManager fragmentManager, int fragmentId) {
-        return (T) fragmentManager.findFragmentById(fragmentId);
-    }
-
-    @SuppressWarnings("unchecked")
-    public static <T> T findFragmentByTag(android.support.v4.app.FragmentManager fragmentManager, String fragmentTag) {
```
Comment (ataulm) [inline src/main/java/com/novoda/notils/caster/Fragments.java]: boom, why you lose these?

File: src/main/java/com/novoda/notils/logger/Novogger.java [hunk 6]
```diff
@@ -1,109 +0,0 @@
-package com.novoda.notils.logger;
-
-import android.content.Context;
-
-public class Novogger {
```
Comment (ataulm) [inline src/main/java/com/novoda/notils/logger/Novogger.java]: :100:

File: src/main/java/com/novoda/notils/caster/Fragments.java (hunk 5 above)
Comment (ouchadam) [inline src/main/java/com/novoda/notils/caster/Fragments.java]: was the only reason we brought in the support library, I think this entire file will probably go

**Comment by ataulm** [comment]
lgtm apart from the dropping of the support fragment methods in the Fragments caster.

File: build.gradle (hunk 2 above)
Comment (ouchadam) [inline build.gradle]: wasn't sure which way the manifests get merged, but 22 is much safer than 23

File: src/main/java/com/novoda/notils/android/logger/Log.java (hunk 4 above)
Comment (ouchadam) [inline src/main/java/com/novoda/notils/android/logger/Log.java]: didn't want to change any of the implementations in this

File: src/main/java/com/novoda/notils/caster/Fragments.java (hunk 5 above)
Comment (blundell) [inline src/main/java/com/novoda/notils/caster/Fragments.java]: didn't mention in the PR desc you removed code! Sneaky

**Comment by blundell** [comment]
The diff really confused me. So much red, where has all the red code gone, been deleted?

Also not sure why the packages need changing for it to be splt into two modules? I'd advise not changing the packages and just doing the module split.

**Comment by ouchadam** [comment]
all the red code is the packages I deleted from the description `Drops the Cursor, Animation, Adapter, Fragment & View Server packages`

**Pull request closed without merging**